app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -411,19 +391,18 @@ class LuciBuildService {
     return true;
   }
 
-  /// Sends a [BuildBucket.scheduleBuild] request using [CheckRunEvent]. It
-  /// returns [true] if it is able to send the scheduleBuildRequest or [false]
-  /// if not.
-  Future<bool> rescheduleUsingCheckRunEvent(CheckRunEvent checkRunEvent) async {
+  /// Sends [ScheduleBuildRequest] for [pullRequest] using [checkRunEvent].
+  /// 
+  /// Returns true if it is able to send the scheduleBuildRequest. Otherwise, false.
+  Future<bool> rescheduleUsingCheckRunEvent(github.PullRequest pullRequest, CheckRunEvent checkRunEvent) async {
     final github.RepositorySlug slug = checkRunEvent.repository!.slug();
     final Map<String, dynamic> userData = <String, dynamic>{};
     final String? commitSha = checkRunEvent.checkRun!.headSha;
```

Comment:
ditto.

Reply:
I think we want to rely on the check run for consistency. There's a chance a user may rerun a check + push a new commit, which case we could result in 2 runs of the latest sha.